pkg/gadget-context/gadget-context.go

```diff
@@ -212,10 +212,26 @@ func (c *GadgetContext) RegisterDataSource(t datasource.Type, name string) (data
 	return ds, nil
 }
 
-func (c *GadgetContext) GetDataSources() map[string]datasource.DataSource {
+func (c *GadgetContext) getDataSources(all bool) map[string]datasource.DataSource {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-	return maps.Clone(c.dataSources)
+
+	ret := maps.Clone(c.dataSources)
+	for name, ds := range ret {
+		// Don't forward unrequested data sources if all is false
+		if !all && !ds.IsRequested() {
+			delete(ret, name)
+		}
+	}
+	return ret
+}
+
+func (c *GadgetContext) GetDataSources() map[string]datasource.DataSource {
+	return c.getDataSources(false)
+}
+
+func (c *GadgetContext) GetDataSourcesAll() map[string]datasource.DataSource {
+	return c.getDataSources(true)
 }
```

Comment:
Mmmm I want the clean `GetDataSources` to return the requested datasources (which is the most common case). Instead, return also the non-requested datasources only if developer intentionally want to include them, so I want that function to be called differently. 

I also thought about adding options to the GetDataSources because maybe in the future we want also to get datasources with specific annotations or tags. But for now, it seems like overkilling the problem. WDYT?

Reply:
Agree, if we see it from the `unreferenced` perspective (like discussed in another comment), it makes more sense to be the default use case to have them filtered. Maybe still `GetDataSourcesAll()` => `GetAllDataSources()` to sound more natural? But I don't mind if you want to keep it the way it is right now.

Reply:
Applied.